cidc_api/auth.py

```diff
@@ -82,8 +82,40 @@ def log_user_and_request_details(is_authorized: bool):
 
         log_user_and_request_details(is_authorized)
 
+        self.enforce_cli_version()
+
         return is_authorized
 
+    def enforce_cli_version(self):
+        """
+        If the current request appears to come from the CLI and not the Portal, enforce the configured
+        minimum CLI version.
+        """
+        user_agent = request.headers.get("User-Agent")
+
+        # e.g., during testing no User-Agent header is supplied
+        if not user_agent:
+            return
+
+        client, version = user_agent.split("/")
+
+        # Old CLI versions don't update the User-Agent header, so we (perhaps dangerously)
+        # assume any request coming from the python requests library is from a "very" old
+        # version of the CLI.
+        is_very_old_cli = client == "python-requests"
```

Comment:
what if someone will use python-requests in their own client impl 😅

Reply:
Hence `perhaps dangerously` 🙂  

Happy to take this out, though.